AMaDda/Screens/Adding/AddingViewController.swift

```diff
@@ -63,22 +68,77 @@ class AddingViewController: UIViewController {
+    }
+    
+    // MARK: - seletor
+    
+    @objc private func didTapAddButton() {
+        guard let text = nickNameTextField.text else { return }
+        print(text)
+        // TODO: 이미지경로(String) 이랑 text(String) 를 array에 append하고 그 array를 UserDefault에 저장
+    }
+    
+    @objc private func didTapProfileImageView(_ gesture: UITapGestureRecognizer) {
+        print("gesture")
+    }
+    
+    @objc private func keyboardWillShow(notification:NSNotification) {
+        if let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
+            UIView.animate(withDuration: 0.2, animations: {
+                self.addButton.transform = CGAffineTransform(translationX: 0, y: -keyboardSize.height + 60)
+            })
+        }
+    }
+    
+    @objc private func keyboardWillHide(notification:NSNotification) {
+        UIView.animate(withDuration: 0.2, animations: {
+            self.addButton.transform = .identity
+        })
+    }
+    
+    // MARK: - function
+    
+    private func setupTapGesture() {
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(didTapProfileImageView(_:)))
+        tapGestureRecognizer.numberOfTapsRequired = 1
+        tapGestureRecognizer.numberOfTouchesRequired = 1
+        profileImageView.addGestureRecognizer(tapGestureRecognizer)
+    }
+    
+    private func setCounter(count: Int) {
+        textFieldLimitLabel.text = "\(count)/10"
+        checkMaxLength(textField: nickNameTextField, maxLength: maxLength)
     }
 
+    private func checkMaxLength(textField: UITextField, maxLength: Int) {
+        if (textField.text?.count ?? 0 > maxLength) {
+            textField.deleteBackward()
+        }
+    }
+    
+    private func setupNotificationCenter() {
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow), name: UIResponder.keyboardWillShowNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
     // MARK: - configure
     private func configureUI() {
         view.backgroundColor = .systemBackground
     }
     
     private func configureAddSubView() {
-        view.addSubview(addingTitleLabel)
-        view.addSubview(profileImageView)
-        view.addSubview(plusIcon)
-        view.addSubview(createNicknameLabel)
-        view.addSubview(nickNameTextField)
-        view.addSubview(underLineView)
-        view.addSubview(textFieldLimitLabel)
-        view.addSubview(addButton)
+        view.addSubviews(addingTitleLabel,
+                         profileImageView,
+                         plusIcon,
+                         createNicknameLabel,
+                         nickNameTextField,
+                         underLineView,
+                         textFieldLimitLabel,
+                         addButton
+                        )
```

Comment:
p5;
요건 완전 취향차이 같은데,
json 파일 생각해보면 괄호 열고 닫기 할 때마다 줄바꿈 다 했던거 같은데
여기 상황에선 괄호 열기닫기 할 때 줄바꿈 안하는건 어떨까요??
뭔가 괄호 하나만 덩그라니 띄워져 있는게...🥲

```suggestion
        view.addSubviews(addingTitleLabel,
                         profileImageView,
                         plusIcon,
                         createNicknameLabel,
                         nickNameTextField,
                         underLineView,
                         textFieldLimitLabel,
                         addButton)
```

Reply:
수정완료!
[deca3a3](https://github.com/DeveloperAcademy-POSTECH/MC3-Team14-HyojaDongHyojadeul/pull/42/commits/deca3a3330ff4aef0d623060e9ed487f8898bf95)